### ISSUE TITLE
ImportError: No module named nets.ColorHandPose3DNetwork

### ISSUE THREAD

**Issue by alxistn** [Prasad9/Classify-HandGesturePose#1]
I get this error "ImportError: No module named nets.ColorHandPose3DNetwork" when I lunch the following command:

    python3 evaluate_pose.py ./pose/test_data

**Comment by vivek-varma** [comment]
I faced the same problem.After checking the forum online,I tried this
create (__init__.py) in nets folder,utils folder and main classify-handgesture folder

![img](https://user-images.githubusercontent.com/25985443/41395151-474ec648-6fca-11e8-889d-43c80f5f37d8.png)

**Comment by abhigarg** [comment]
I tried creating __ init __ py file inside nets folder but still I am getting the same error. I added import ColorHandPose3DNetwork inside this python file

**Comment by vivek-varma** [comment]
@abhigarg did u create init.py in utils and root folder?
and also __init__.py must be empty.

**Comment by Prasad9** [comment]
Hi @alxistn , first of all I am very sorry that I didn't reply to you in a timely manner.
The reason why you are getting this error is that you have to download the original weights file from Hand3d Project. I have mentioned it in my ReadMe. Maybe I should have given the link to download it as well over there.
The link to download is this: https://lmb.informatik.uni-freiburg.de/projects/hand3d/ColorHandPose3D_data_v3.zip

However the above downloaded folder will contain three folders. Among those three, paste the folder named as `weights` into your project folder. 
Hope this answer helps you! Please can you close the issue if your issue has been resolved.

**Comment by JasOlean** [comment]
Hi,  when I run the run.py file, I got this error although there are pickle files in weights folder.
nets/ColorHandPose3DNetwork.py", line 52, in init
    assert os.path.exists(file_name), "File not found."
AssertionError: File not found.

**Comment by Baakchsu** [comment]
I downloaded the weights file and put it in the project folder but now it throws a CUBLAS error.
This is the error 
"_device.cc:1312] Adding visible gpu devices: 0
2018-09-21 17:15:45.865543: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 4915 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1060 6GB, pci bus id: 0000:01:00.0, compute capability: 6.1)
Loaded 32 variables from ./weights/handsegnet-rhd.pickle
Loaded 102 variables from ./weights/posenet3d-rhd-stb-slr-finetuned.pickle
2018-09-21 17:15:51.507728: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_blas.cc:443] failed to create cublas handle: CUBLAS_STATUS_ALLOC_FAILED"
Could you please give any information about the 3DHand model's compatibility with windows ?

**Comment by shujathkhan** [comment]
> I downloaded the weights file and put it in the project folder but now it throws a CUBLAS error.
> This is the error
> "_device.cc:1312] Adding visible gpu devices: 0
> 2018-09-21 17:15:45.865543: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 4915 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1060 6GB, pci bus id: 0000:01:00.0, compute capability: 6.1)
> Loaded 32 variables from ./weights/handsegnet-rhd.pickle
> Loaded 102 variables from ./weights/posenet3d-rhd-stb-slr-finetuned.pickle
> 2018-09-21 17:15:51.507728: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_blas.cc:443] failed to create cublas handle: CUBLAS_STATUS_ALLOC_FAILED"
> Could you please give any information about the 3DHand model's compatibility with windows ?

Hi Baakchsu,

I faced the same problem earlier with Nvidia GTX , nut the error was CUDNN_STATUS_ALLOC_FAILED,
I simply updated my Nvidia driver. It worked out.
Try updating or reinstalling your graphics drivers.

**Comment by maciaspeter** [comment]
1. create a `__init__.py` file in the root folder
2. execute this command - `export PYTHONPATH=$PYTHONPATH:pwd:pwd/slim`

**Comment by pavankumar3175** [comment]
> Hi @alxistn , first of all I am very sorry that I didn't reply to you in a timely manner.
> The reason why you are getting this error is that you have to download the original weights file from Hand3d Project. I have mentioned it in my ReadMe. Maybe I should have given the link to download it as well over there.
> The link to download is this: https://lmb.informatik.uni-freiburg.de/projects/hand3d/ColorHandPose3D_data_v3.zip
> 
> However the above downloaded folder will contain three folders. Among those three, paste the folder named as `weights` into your project folder.
> Hope this answer helps you! Please can you close the issue if your issue has been resolved.

I have downloaded the weights and copied into my Project location. Still I am facing the same issue.

**Comment by pavankumar3175** [comment]
> Hi, when I run the run.py file, I got this error although there are pickle files in weights folder.
> nets/ColorHandPose3DNetwork.py", line 52, in init
> assert os.path.exists(file_name), "File not found."
> AssertionError: File not found.

Yup even i am facing the same issue, Did you get any solution to this.

**Comment by akashj0070** [comment]
Classify-HandGesturePose-master\nets\ColorHandPose3DNetwork.py", line 23, in <module>
    from utils.general import *
ModuleNotFoundError: No module named 'utils.general'


i am getting this error

**Comment by grgvsrdvsdFva** [comment]
Hi I am a student from Hong Kong
According to what I have observed, the 'nets' folder collides with a package called "NETS" by PyPi
Just simply move the ColourHandPose3DNetwork.py file out of the "nets" file and directly import it.
Like:
![image](https://user-images.githubusercontent.com/102578924/160555007-971d683e-f176-4aaa-b7d1-71b592810234.png)
Hope this helps you
